src/Tasks/SystemState.cs

```diff
@@ -27,6 +27,8 @@ namespace Microsoft.Build.Tasks
     [Serializable]
     internal sealed class SystemState : StateFileBase, ISerializable
     {
+        private Hashtable upToDateLocalFileStateCache = new Hashtable();
```

Comment:
Why not make this a Dictionary?

Reply:
Consistency. There's very little gain in terms of perf,